sympy/core/function.py

```diff
@@ -621,8 +621,11 @@ def _eval_is_commutative(self):
         return fuzzy_and(a.is_commutative for a in self.args)
 
     def _eval_is_meromorphic(self, x, a):
+        from sympy.functions.elementary.piecewise import Piecewise
         if not self.args:
             return True
+        if isinstance(self, Piecewise) and any((arg.has(x) or arg[0].is_number) for arg in self.args[1:]):
```

Comment:
Why should arg in self.arg[1:] not have x? A univariate Piecewise function may have x in each argument.

Reply:
This method if for general functions that may have several arguments. To say that such a function is meromorphic means that it is meromorphic in the first argument. This does not work with Piecewise functions. There should be a `Piecewise.is_meromorphic` that would invoke `_eval_is_meromorphic` for the piece specified by `a`.

Reply:
Yeah True , this was just a low level implementation and I expecting that someone points out that `Piecewise.is_meromorphic` , should be implemented . I'll try this out and get back soon !